codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/error/CombinedErrorGenerator.kt

```diff
@@ -43,6 +46,13 @@ fun OperationShape.errorSymbol(symbolProvider: SymbolProvider): RuntimeType {
     return RuntimeType("${symbol.name}Error", null, "crate::error")
 }
 
+fun OperationShape.eventStreamErrorSymbol(symbolProvider: SymbolProvider, target: CodegenTarget): RuntimeType {
+    val symbol = symbolProvider.toSymbol(this)
+    return when (target) {
+        CodegenTarget.CLIENT -> RuntimeType("${symbol.name}ClientEventStreamError", null, "crate::error")
+        CodegenTarget.SERVER -> RuntimeType("${symbol.name}ServerEventStreamError", null, "crate::error")
```

Comment:
I don't think we need to differentiate between client/server in the error names. If you disagree, I want to understand better why this is required.

Separately, we should brainstorm some shorter names for these 😄

Reply:
Not doing so leads to the same problem of having all of them together (operation and event streams), that the client has the option to yield errors it's not supposed to. Merging them would generate like
```
        let payload = match input.kind {
            crate::error::CapturePokemonOperationEventStreamErrorKind::MasterBallUnsuccessful(inner) => ...
            crate::error::CapturePokemonOperationEventStreamErrorKind::Unhandled(_inner) => ...
          _ => ???
```
If they are merged together, at this point it's the same to having all of them under the operation errors. What do you think?
As for the names, any suggestion I'm happy to take (input/output as well) :)

Reply:
Implemented, thank you